### PR TITLE
ScreenOrientation support

### PR DESCRIPTION
**Pull request by toastal**
W3 spec for `ScreenOrientation`: https://drafts.csswg.org/cssom-view/

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,8 @@
...
-/bower_components
-/node_modules
-/output
+/bower_components/
+/node_modules/
+/output/
+/dce-output/
 /generated-docs
+.spago/
+.purs-repl
```
Comment (keijokapp) [inline .gitignore]: 1. I'm not sure, but I think that `.purs-repl` and `dce-output` shouldn't be in shared `.gitignore`. These seem to be specific to a particular development environment and/or developer. I recommend adding such kind of files to `.git/info/exclude`.

1. `.gitignore` entries should start with a `/` whenever possible. That way, we don't have to worry about recursive matches and confusion that might come with it.

2. I'd prefer to _not_ have a trailing `/` at the end of any `.gitignore` entry. There's no relevant benefits for a slash, except maybe for documenting the tree structure but I don't think anyone needs that documentation.

File: src/Web/CSSOMView/ScreenOrientation.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/* Web.CSSOMView.ScreenOrientation */
```
Comment (keijokapp) [inline src/Web/CSSOMView/ScreenOrientation.js]: That comment is unneccessary. The module name is already known from the file name.

File: src/Web/CSSOMView/ScreenOrientation.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+"use strict";
+
+exports.lockImpl = function lock(orientation) {
+  return function(so) {
+    return function(onError, onSuccess) {
+      so.lock(orientation).then(onSuccess).catch(onError);
```
Comment (keijokapp) [inline src/Web/CSSOMView/ScreenOrientation.js]: A small nitpick, but `.then` has a 2-argument variant for exactly this use case. The code would be
```js
so.lock(orientation).then(onSuccess, onError);
```

**Review by keijokapp — CHANGES_REQUESTED** [review]
Thanks for the PR!

Sorry for not reviewing that earlier. I haven't used purescript for over a year and don't see myself using it within foreseeable future. I don't have a test project to test these changes. I guess they are okay.

One big question is about the addition of Spago setup. Seems that Spago is still used only for applications, not libraries like this one. I looked at some other purescript-web packages and they don't have Spago setup. So I guess, you should remove it from the PR.

**Comment by toastal** [comment]
Updated. And I had personally tested these changes in a staging environment at a previous contract for what I presume could be future production use via the fork I've made.

**Comment by keijokapp** [comment]
Thank you!

**Pull request merged**